.github/workflows/dispatch-build-bottle.yml

```diff
@@ -0,0 +1,107 @@
+name: Build bottles from 'brew request-bottle' trigger
+
+on:
+  repository_dispatch:
+    types:
+      - 'Bottle *'
+
+jobs:
+  bottle:
+    runs-on: ${{github.event.client_payload.macos}}
+    env:
+      HOMEBREW_DEVELOPER: 1
+      HOMEBREW_FORCE_HOMEBREW_ON_LINUX: 1
+      HOMEBREW_NO_ANALYTICS: 1
+      HOMEBREW_NO_AUTO_UPDATE: 1
+      GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED: 1
+      GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+    timeout-minutes: 1440
+    steps:
+      - name: Print details
+        run: |
+          echo macos=${{github.event.client_payload.macos}}
+          echo formula=${{github.event.client_payload.formula}}
+          echo issue=${{github.event.client_payload.issue}}
+
+      - name: Update Homebrew
+        run: brew update-reset
+
+      - name: Setup Git
+        uses: Homebrew/actions/git-user-config@master
+
+      - name: Set up Git repository
+        run: |
+          cd $(brew --repo ${{github.repository}})
+          git clean -ffdx
+          git fetch --prune --force origin master
+          git fetch --prune --force origin ${{github.sha}}
+          git checkout --force ${{github.sha}}
+          git log -1
+
+      - name: Run brew test-bot --only-cleanup-before
+        run: brew test-bot --only-cleanup-before
+
+      - name: Run brew test-bot --only-setup
+        run: brew test-bot --only-setup
+
+      - name: Run brew test-bot --only-formulae
+        run: |
+          rm -rf bottles
+          mkdir bottles
+          cd bottles
+          brew test-bot --only-formulae --keep-old ${{github.event.client_payload.formula}}
```

Comment:
I think ideally `--keep-old` would be conditional on the payload. This is fine for a first version, though!